gym/vector/utils/spaces.py

```diff
@@ -88,13 +108,14 @@ def _batch_space_dict(space, n=1):
                 (key, batch_space(subspace, n=n))
                 for (key, subspace) in space.spaces.items()
             ]
-        )
+        ),
+        seed=copy(space.np_random),
     )
 
 
 @batch_space.register(Space)
 def _batch_space_custom(space, n=1):
-    return Tuple(tuple(space for _ in range(n)))
+    return Tuple(tuple(space for _ in range(n)), seed=copy(space.np_random))
```

Comment:
I believe this change here introduces a subtle bug:
Each subspace is using the a Generator with the same state for each subspace, which I think will lead to each space giving back identical samples.
Correct me if I'm wrong about this, (preferably with a unit test :wink: )!

Reply:
This is a tricky one, but I dont think so. Looking at the Tuple code, then the seed really doesn't do anything except if you are trying to set a new seed
As the Tuple uses the same space and the Tuple sample uses the subspace sample iteratively which updates each time, Tuple shouldn't produce the same random samples.
Im adding Space to the batch space tests

Reply:
If you're only doing a shallow copy, and all the individual subspaces will actually share the same Generator, then yeah, as you describe (I think),  they will all produce different samples. However, that's quite weird behaviour IMO. I believe all the subspaces should be independently sampled, and each of them should produce different results. It's unclear to me how that could be achieved with a Generator, but something I'd do would probably look like this with `np.random.RandomState`:
```python
def batch_space(space: Space, n: int) -> Space:
    subspaces = [copy.deepcopy(space) for _ in range(n)]
    if space.rng:
        rng: np.random.RandomState = space.rng
        # Create n random states from the original random state.
        seeds = space.rng.integers(0, 1e8, n)
        for subspace, seed in zip(subspaces, seeds):
            space.rng = np.random.RandomState(seed)
    return spaces.Tuple(subspaces)
```

Reply:
There are two possible issues
1. the original space and the batched space random number generator are the same object meaning that sampling one space effects the other
2. The subspaces of the batched space produce the same random values on each sample. i.e. space.sample = [1], batched_space.sample = [1, 1, 1]

So I believe that I have solved both issues
For issue 1, I have made a new update where we just do a deepcopy of the original space to prevent the two spaces sharing the same random number generator
 ```
def batch_space(space: Space, n: int = 1) -> Space:
    # Without deepcopy, then the space.np_random is batched_space.spaces[0].np_random
    # Which is an issue if you are sampling actions of both the original space and the batched space
    space_copy = deepcopy(space)
    return Tuple(tuple(space_copy for _ in range(n)), seed=copy(space.np_random))
```

For issue 2, this issue is more complex but this issue has never been an issue due to the way that the tuple sample function works. As the subspaces all share the same random number generator, as each subspace is the same object, and as the tuple sample function is `return tuple(space.sample() for space in self.spaces)`. Then there isn't an issue as the space random number generator is updated each time sample is called. Meaning that the sample always produce different values

Does that address your issue or is it different?

Reply:
Thanks for the clarification, however this doesn't really address the second issue. The subspaces shouldn't have implicitly-shared RNG. They should each have independent (and different!) RNG that are deterministic with respect to the base space.

Again, here's the solution:
Sample N seeds with the rng of the base space, and then pass these N seeds to each subspace (or a Generator for each subspace).

Reply:
Changed in 4fdd037c6531e01fe4e003eae09c357f44778187